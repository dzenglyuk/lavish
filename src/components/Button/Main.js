import React from "react";
import "./Main.css";

const STYLES = ["btn--primary", "btn--outline"];
const SIZES = ["btn--medium", "btn--large", "btn--modile", "btn--wide"];
const COLORS = ["primary", "blue", "red", "green"];

const Button = ({ children, type, onClick, btnStyle, btnSize, btnColor }) => {
    const checkProp = (propName, propValue, name) => {
		let defaultProp = propName[0];
		if (name && name === 'color') defaultProp = '';
		return propName.includes(propValue) ? propValue : defaultProp;
    };
    
    return (
        <button
            className={`btn 
			${checkProp(STYLES, btnStyle)} 
			${checkProp(SIZES, btnSize)} 
			${checkProp(COLORS, btnColor, 'color')}`}
            onClick={onClick}
            type={type}
        >
            {children}
        </button>
    );
};

Button.defaultProps = {
    btnStyle: "btn--primary",
    btnSize: null,
    btnColor: null
}

export default React.memo(Button);
