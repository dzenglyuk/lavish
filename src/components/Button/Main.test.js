import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { StaticRouter as Router } from "react-router";
import userEvent from "@testing-library/user-event";

import Button from "./Main";

test("Renders correctly", () => {
    const { getByText } = render(
        <Button> Test Button </Button>
    );

    getByText("Test Button");
    expect(getByText("Test Button").className).toMatch(/btn--medium/);
    expect(getByText("Test Button").className).toMatch(/btn--primary/);
});

test("Handles click", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
        <Button handleClick={handleClick}> Test Button </Button>
    );
    userEvent.click(getByText("Test Button"));
    expect(handleClick).toHaveBeenCalledTimes(1);
});

test("Matches the snapshot", () => {
    const tree = renderer
        .create(
            <Button> Test Button </Button>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});