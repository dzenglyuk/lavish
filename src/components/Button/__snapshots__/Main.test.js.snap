// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<button
  className="btn 
			btn--primary
			btn--medium
			"
>
   Test Button 
</button>
`;
